peps/pep-0810.rst

```diff
@@ -0,0 +1,1345 @@
+
+* In addition, it is useful to provide a mechanism to activate or deactivate lazy
+  imports at a global level. While the primary design centers on explicit syntax,
+  there are scenarios—such as large applications, testing environments, or
+  frameworks—where enabling laziness consistently across many modules provides
+  the most benefit. A global switch makes it easy to experiment with or enforce
+  consistent behavior, while still working in combination with the filtering API
+  to respect exclusions or tool-specific configuration. This ensures that global
+  adoption can be practical without reducing flexibility or control.
+
+
+Specification
+=============
+
+Grammar
+-------
+
+A new soft keyword ``lazy`` is added. A soft keyword is a context-sensitive keyword
+that only has special meaning in specific grammatical contexts; elsewhere it can be
+used as a regular identifier (e.g., as a variable name). The ``lazy`` keyword only
+has special meaning when it appears before import statements:
+
+.. code-block:: text
+
+  import_name:
+      | 'lazy'? 'import' dotted_as_names
+
+  import_from:
+      | 'lazy'? 'from' ('.' | '...')* dotted_name 'import' import_from_targets
+      | 'lazy'? 'from' ('.' | '...')+ 'import' import_from_targets
+
+Syntax restrictions
+~~~~~~~~~~~~~~~~~~~
+
+The soft keyword is only allowed at the global (module) level, **not** inside
+functions, class bodies, with ``try``/``with`` blocks, or ``import *``. Import
+statements that use the soft keyword are *potentially lazy*. Imports that
+can't be lazy are unaffected by the global lazy imports flag, and instead
+are always eager.
+
+Examples of syntax errors:
+
+.. code-block:: python
+
+  # SyntaxError: lazy import not allowed inside functions
+  def foo():
+      lazy import json
+
+  # SyntaxError: lazy import not allowed inside classes
+  class Bar:
+      lazy import json
+
+  # SyntaxError: lazy import not allowed inside try/except blocks
+  try:
+      lazy import json
+  except ImportError:
+      pass
+
+  # SyntaxError: lazy import not allowed inside with blocks
+  with suppress(ImportError):
+      lazy import json
+
+  # SyntaxError: lazy from ... import * is not allowed
+  lazy from json import *
+
+Semantics
+---------
+
+When the ``lazy`` keyword is used, the import becomes *potentially lazy*.
+Unless lazy imports are disabled or suppressed (see below), the module is
+not loaded immediately at the import statement; instead, a lazy proxy object
+is created and bound to the name. The actual module is loaded on first use
+of that name.
+
+Example:
+
+.. code-block:: python
+
+  import sys
+
+  lazy import json
+
+  print('json' in sys.modules)  # False - module not loaded yet
+
+  # First use triggers loading
+  result = json.dumps({"hello": "world"})
+
+  print('json' in sys.modules)  # True - now loaded
+
+A module may contain a ``__lazy_modules__`` attribute, which is a list of
```

Comment:
How does this interact with from imports? If I have `__lazy_modules__ = ["a"]`, does that make both `from a import b` and `from a import c` automatically lazy? How about `import a.b` or `from a.b import c`?

Also, does it have to be a list or can it be any sequence?